include/fmt/format.h

```diff
@@ -83,7 +83,9 @@
 #  if FMT_CPLUSPLUS >= 202002L
 #    if FMT_HAS_CPP_ATTRIBUTE(no_unique_address)
 #      define FMT_NO_UNIQUE_ADDRESS [[no_unique_address]]
-#    elif FMT_MSC_VERSION >= 1929  // VS2019 v16.10 and later
+// VS2019 v16.10 and later except clang-cl (https://reviews.llvm.org/D110485)
+#    elif __has_cpp_attribute(msvc::no_unique_address) || \
```

Comment:
__has_cpp_attribute -> FMT_HAS_CPP_ATTRIBUTE

Reply:
Actually, I don't think we need an attribute check since the branch below only handles specific compiler (versions).

Reply:
OK, `FMT_HAS_CPP_ATTRIBUTE` done, missed that.

The reason I put the `msvc::no_unique_address` attr check is to future proof if support gets added to clang-cl, rather than always not setting.

I suppose if clang-cl waits to enable the standard `no_unique_address` until after cl does and never bothers with the `msvc` attr it'll be moot, but this way at least it won't need revising either way... Or at least that was the intention!

Reply:
I don't think we should be adding support for some not-standard attribute that may or may not be supported in the future. Let's remove the attribute check.

Reply:
OK sure, done.

Seems a shame to miss out on the size optimization if it does get added, though! Not that I've looked at where it's used in fmtlib 😀